src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -62,18 +90,9 @@ export class NavigationPlugin implements Plugin {
         if (performance.getEntriesByType(NAVIGATION).length === 0) {
             this.performanceNavigationEventHandlerTimingLevel1();
         } else {
-            const navigationObserver = new PerformanceObserver((list) => {
-                list.getEntries().forEach((event) => {
-                    if (event.entryType === NAVIGATION) {
-                        this.performanceNavigationEventHandlerTimingLevel2(
-                            event
-                        );
-                    }
-                });
-            });
-            navigationObserver.observe({
-                entryTypes: [NAVIGATION]
-            });
+            this.performanceNavigationEventHandlerTimingLevel2(
+                performance.getEntriesByType(NAVIGATION)[0]
```

Comment:
Is it guaranteed that there is only one navigation timing entry?

Reply:
Good catch, I will change it to iterate instead.

Reply:
Actually, just found this: https://www.w3.org/TR/navigation-timing-2/ 
"Only the [current document](https://www.w3.org/TR/navigation-timing-2/#dfn-current-document) resource is included in the performance timeline; there is only one [PerformanceNavigationTiming](https://www.w3.org/TR/navigation-timing-2/#dom-performancenavigationtiming) object in the performance timeline."